#!/usr/bin/env python
PACKAGE = "aruco_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("min_image_size", double_t, 0, "minimum size of marker as a % of image area", 0.02, 0.0, 1.0)
gen.add("normalizeImage", bool_t, 0, "normalizeImage", True)
gen.add("dctComponentsToRemove", int_t, 0, "DCT components to remove", 2, 1, 4)
gen.add("degree", int_t, 0, "Degree to rotate", 0, 0, 360)
detection_mode_enum = gen.enum([    gen.const("Normal",      int_t, 0, ""),
                                    gen.const("Fast",     int_t, 1, ""),
                                    gen.const("Video_Fast",      int_t, 2, "")],
                     "detection_mode enum")

gen.add("detection_mode", int_t, 0, "The detection mode , affects speed and reliability", 1, 0, 2, edit_method=detection_mode_enum)
exit(gen.generate(PACKAGE, "double", "ArucoThreshold"))
